terraform/modules/security-groups/main.tf

```
@@ -0,0 +1,77 @@
+############################
+# Security‑group shells
+############################
+
+resource "aws_security_group" "load_balancer" {
+  name        = "${var.prefix}-alb-sg"
+  description = "Allow HTTP from the Internet to ALB"
+  vpc_id      = var.vpc_id
+
+  tags = {
+    Name = "${var.prefix}-alb-sg"
+  }
+}
+
+resource "aws_security_group" "app" {
+  name        = "${var.prefix}-ec2-sg"
+  description = "Allow SSH from admin CIDR and HTTP (8080) only from ALB"
+  vpc_id      = var.vpc_id
+
+  tags = {
+    Name = "${var.prefix}-ec2-sg"
+  }
+}
+
+############################
+# ALB rules
+############################
+
+# Ingress : HTTP 80 from anywhere
+resource "aws_vpc_security_group_ingress_rule" "load_balancer_http_in" {
```

```suggestion
resource "aws_vpc_security_group_ingress_rule" "load_balancer_http" {
```
「Terraformのリソース名にサービス名を入れない」というのは冗長的な表現になるためです。これは他にも流用できる考えであり、こちらのリソースではingressとinが被っており、冗長的な表現となります！